update LatestInLibraryView to use getLatestMedia

elliots opened the pull request:
This now matches what the web player does.

<img width="1510" alt="image" src="https://github.com/jellyfin/Swiftfin/assets/622455/ff378c51-859c-4091-bfc4-c1c4db5ec729">

Before this change, it displayed the latest *shows* added.

<img width="895" alt="image" src="https://github.com/jellyfin/Swiftfin/assets/622455/e7205dbb-f4d9-490f-9a4d-9f9ce6b08bac">

elliots commented:
I've removed the trailing `SeeAllPosterButton`, as it wasn't happy with the new `LatestInLibraryViewModel`. But I'm sure I can work that out.

elliots commented:
The `getLatestMedia` doesn't appear to support paging, so I'm grabbing 3x the number so the trailing "Show All" button has a bit more in it. Not very nice. Is there a better way?

LePips commented:
This breaks the "random" selection. I admit I didn't take a good look at how the parameter creation was refactored when that work was done but it should be redone.

elliots commented:
I've fixed the linting issues, and implemented `getRandomItemFromLibrary`. The implementation grabs a random item from the lists and wraps it in `BaseItemDtoQueryResult`. Also not brilliant but i think better than getting a random one from the whole library. 

Is using `JellyfinAPIError` if it hasn't been loaded yet ok? 

I couldn't see where the toolbar with the `RandomItemButton` is displayed, so I haven't actually tested it yet.

LePips commented on Shared/ViewModels/LatestInLibraryViewModel.swift:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    override public func getRandomItemFromLibrary() async throws -> BaseItemDtoQueryResult {
+        if items.isEmpty {
```
Sadly I let this function pass as is, since it should just return the item, but this just needs:
```swift
BaseItemDtoQueryResult(items: items.elements)
```

since right now the button handles choosing a random item. I will change that.

elliots replied:
Done...

LePips replied:
Looks like you still need to lint.

elliots replied:
Done
